include/ginkgo/core/base/array.hpp

```diff
@@ -366,6 +383,54 @@ class Array {
         return *this;
     }
 
+    /**
+     * Copies and converts data from another array with another data type.
+     * In the case of an array target, the array is resized to match the
+     * source's size. In the case of a view target, if the dimensions are not
+     * compatible a gko::OutOfBoundsError is thrown.
+     *
+     * This does not invoke the constructors of the elements, instead they are
+     * copied as POD types.
+     *
+     * The executor of this is preserved. In case this does not have an assigned
+     * executor, it will inherit the executor of other.
+     *
+     * @param other  the Array to copy from
+     * @tparam OtherValueType  the value type of `other`
+     *
+     * @return this
+     */
+    template <typename OtherValueType>
+    xstd::enable_if_t<!std::is_same<ValueType, OtherValueType>::value, Array>
+        &operator=(const Array<OtherValueType> &other)
```

Comment:
Could you please also add a `operator=(Array<OtyerValueType> &&)` version?

Reply:
Can you give a use-case for this operator (i.e. where providing a second overload gives us an advantage?) or how the move-assignment should differ from copy-assignment?

Reply:
With this operator I wasn't sure myself. At the time I wrote the comment, I simply thought: We support a `move` where we also support `operator=`, so the `move` is missing here.
I am actually fine if we skip the `move` here because we can never really reuse anything, so moving would not have any advantage over the copying.